src/components/motor/motor.hpp

```diff
@@ -0,0 +1,137 @@
+/// @file components/motor/motor.hpp
+///
+/// @brief Defines a `Motor` component.
+#pragma once
+
+#include <common/proto_type.hpp>
+#include <common/utils.hpp>
+#include <component/motor/v1/motor.pb.h>
+#include <config/resource.hpp>
+#include <registry/registry.hpp>
+#include <string>
+#include <subtype/subtype.hpp>
+
+namespace viam {
+namespace cppsdk {
+
+/// @defgroup Motor Classes related to the `Motor` component.
+
+/// @class MotorSubtype
+/// @brief Defines a `ResourceSubtype` for the `Motor` component.
+/// @ingroup Motor
+class MotorSubtype : public ResourceSubtype {
+   public:
+    std::shared_ptr<ResourceServerBase> create_resource_server(
+        std::shared_ptr<SubtypeService> svc) override;
+    std::shared_ptr<ResourceBase> create_rpc_client(std::string name,
+                                                    std::shared_ptr<grpc::Channel> chan) override;
+    MotorSubtype(const google::protobuf::ServiceDescriptor* service_descriptor)
+        : ResourceSubtype(service_descriptor){};
+};
+
+/// @class Motor motor.hpp "components/motor/motor.hpp"
+/// @brief A `Motor` represents physical hardware that controls the rotation of an axle
+/// @ingroup Motor
+///
+/// This acts as an abstract base class to be inherited from by any drivers representing
+/// specific motor implementations. This class cannot be used on its own.
+class Motor : public ComponentBase {
+   public:
+    /// @struct position
+    /// Current position of the motor relative to its home
+    typedef double position;
+
+    /// @struct power_status
+    /// @brief Information about power-state of the motor
+    struct power_status {
+        /// True if the motor is on
+        bool is_on;
+        /// Power percent (from 0 to 1, or from -1 to 1 for motors that support negative
+        /// power), based on the last command sent to motor. If the last command was a stop command,
+        /// this value will be 0.
+        double power_pct;
+    };
+
+    /// @struct properties
+    /// @brief Features that the specific motor supports
+    struct properties {
+        /// True if the motor supports reporting its position
+        bool position_reporting;
```

Comment:
Yes, it is because of how properties is defined. Properties is defined to be the set of all options that this motor supports. I think it would be unwise for us to assume that there will only ever be one feature that a motor supports.